protocol/messenger_share_urls.go

```diff
@@ -446,28 +446,31 @@ func (m *Messenger) prepareEncodedUserData(contact *Contact) (string, string, er
 		return "", "", err
 	}
 
-	userProto := &protobuf.User{
-		DisplayName: contact.DisplayName,
-		Description: contact.Bio,
-	}
+	var encodedData = ""
+	if contact.DisplayName != "" || contact.Bio != "" {
+		userProto := &protobuf.User{
+			DisplayName: contact.DisplayName,
+			Description: contact.Bio,
+		}
 
-	userData, err := proto.Marshal(userProto)
-	if err != nil {
-		return "", "", err
-	}
+		userData, err := proto.Marshal(userProto)
+		if err != nil {
+			return "", "", err
+		}
 
-	urlDataProto := &protobuf.URLData{
-		Content: userData,
-	}
+		urlDataProto := &protobuf.URLData{
+			Content: userData,
+		}
 
-	urlData, err := proto.Marshal(urlDataProto)
-	if err != nil {
-		return "", "", err
-	}
+		urlData, err := proto.Marshal(urlDataProto)
+		if err != nil {
+			return "", "", err
+		}
 
-	encodedData, err := encodeDataURL(urlData)
-	if err != nil {
-		return "", "", err
+		encodedData, err = encodeDataURL(urlData)
+		if err != nil {
+			return "", "", err
+		}
```

Comment:
This can be simplified by just checking the `urlData == ""` here.

```suggestion
	userProto := &protobuf.User{
		DisplayName: contact.DisplayName,
		Description: contact.Bio,
	}

	userData, err := proto.Marshal(userProto)
	if err != nil {
		return "", "", err
	}

	urlDataProto := &protobuf.URLData{
		Content: userData,
	}

	urlData, err := proto.Marshal(urlDataProto)
	if err != nil {
		return "", "", err
	}
	
	if urlData == "" {
	    return "", "", nil
	}

	encodedData, err := encodeDataURL(urlData)
	if err != nil {
		return "", "", err
	}
```